### ISSUE TITLE
Two identical requests using the same cassette fail

### ISSUE THREAD

**Issue by wizardone** [vcr/vcr#558]
Hey,

I have a rather shitty task of creating invoices in a third party service. In our system whenever a customer creates a booking of some kind two identical requests should be made to this external service(yes, i know this callback thingy is far from optimal, but unfortunately i cannot change the way this behaves). However it seems vcr does not handle this properly. I get a `VCR::Errors::UnhandledHTTPRequestError` every time. So i have something like this (example is just to illustrate)

```ruby
class Booking
after_create do
  generate_first_invoice
  generate_second_invoice
end
end
```
```ruby
let(:booking) do
  VCR.use_cassette('create_invoice') do
    create(:booking)
  end
end
```
I tried commenting out one of the external calls in the `after_create` and vcr started working as expected. Maybe i'm missing something simple here, i don't think that i need to use `:new_episodes` or something like this?

**Comment by mcfiredrill** [comment]
Does it record the cassette successfully the first time you run the test?

Can you share what the actual requests look like? 
Are there any query params being used? 
Are you absolutely *sure* the two requests are exactly the same each time?

**Comment by wizardone** [comment]
Yes, the cassette is recorded successfully the first time. Thanks to you @mcfiredrill i reminded myself that the request are only 99% identical and they need to be 100% 

The *only* difference in the request is the value of `customer_id` that i pass, which obviously caused the issue.
So what i did is i recorded a request for each of the invoices and then use them like this:
```ruby
let(:booking) do
    VCR.use_cassette('create_first_invoice') do
      VCR.use_cassette('create_second_invoice') do
        create(:customer_booking)
      end
    end
  end
```
Not very pretty, but it is working.
Sorry to bother you with this, did not pay enough attention to the actual params, that were being passed. Many thanks for your help!
